b_cfn_elasticsearch_cloner/cloner_source/index.py

```diff
@@ -51,52 +56,84 @@ def handler(event: Dict[Any, Any], context: Any) -> None:
     :return: No return.
     """
 
-    logger.info("Starting processing DynamoDB events.")
+    logger.info('Starting processing DynamoDB events.')
+
+    use_embeddings = bool(
+        SAGEMAKER_EMBEDDINGS_KEY and SAGEMAKER_ENDPOINT_NAME and EMBEDDINGS_KEY)
+
+    if not use_embeddings and any([SAGEMAKER_EMBEDDINGS_KEY, SAGEMAKER_ENDPOINT_NAME, EMBEDDINGS_KEY]):
+        raise OSError(
+            f'To use sentence embedding, all of the following enviroment variables are required: '
+            f'SAGEMAKER_ENDPOINT_NAME, SAGEMAKER_EMBEDDINGS_KEY, EMBEDDINGS_KEY'
+        )
 
     # Send data to elasticsearch using bulk API.
     succeeded, failed = bulk(
         es,
-        dynamodb_to_es_generator(event),
+        dynamodb_to_es_generator(event, use_embeddings),
         stats_only=True,
         raise_on_error=False,
         raise_on_exception=False,
     )
 
-    logger.info(f"Finished processing DynamoDB events. Succeeded: {succeeded}, failed: {failed}")
+    logger.info(f'Finished processing DynamoDB events. Succeeded: {succeeded}, failed: {failed}')
+
+
+def get_sm_features(source: str, sagemaker_client: Any, endpoint_name: str):
+    sagemaker_response = sagemaker_client.invoke_endpoint(
+        EndpointName=endpoint_name,
+        Body=source.encode(encoding='utf-8'),
+        ContentType='text/plain'
+    )
+    source_features = json.loads(sagemaker_response['Body'].read().decode())
+    return source_features
+
+
+def add_embedding(item: Dict) -> Dict:
```

Comment:
with this function you're not "adding" an embedding. What you are doing here is "getting the embeddings for an item", thus better function name would be something like `get_embeddings` or smth like that

Reply:
if this function only needs a question from `item['question']` don't give the whole dict as an input. Make this function more modular by requesting input text directly.